paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -313,6 +341,75 @@ internal class IntentConfirmationHandler(
         )
     }
 
+    private fun launchGooglePay(
+        googlePay: PaymentConfirmationOption.GooglePay,
+        arguments: Args,
+    ) {
+        if (googlePay.config.merchantCurrencyCode == null && !arguments.initializationMode.isProcessingPayment) {
+            val message = "GooglePayConfig.currencyCode is required in order to use " +
+                "Google Pay when processing a Setup Intent"
+
+            logger.warning(message)
+
+            onIntentResult(
+                Result.Failed(
+                    cause = IllegalStateException(message),
+                    message = R.string.stripe_something_went_wrong.resolvableString,
+                    type = ErrorType.Internal,
+                )
+            )
+
+            return
+        }
+
+        _state.value = State.Confirming(contentVisible = false)
```

Comment:
what does this `contentVisible` variable represent? Do we use this to hide payment sheet?

Reply:
We use this to hide `PaymentSheet` when the Google Pay sheet is shown. This is not great modeling and is only relevant for the `PaymentSheet`/`Google Pay` use case. I had re-modeling in this PR but ultimately chose to remove it given the size. Going to follow up with a change in modeling!

Reply:
sounds good! Do you think you could also update the name of this when you do to be something more specific? e.g. `isPaymentSheetVisible`? It's not super obvious what "content" this variable refers to. (doing this in a follow up sgtm if you are going to update the modeling anyways)

Reply:
Can rename in this PR!

Reply:
ty!!